app/src/viewmodels/camera/CameraViewModel.tsx

```diff
@@ -99,15 +101,31 @@ export default function useCameraViewModel(
   // Create the challenge
   const makeChallenge = async () => {
     try {
+      // Check if location is enabled when creating a challenge in a group
+      if (group_id != "" && group_id != "home") {
+        if (!isLocationEnabled || location == null) {
+          alert("You need to enable location to create a challenge in a group");
+          navigation.navigate("GroupScreen", { currentGroup: group_id });
+          return;
+        }
+
+        // Check if the location is within the group's area
+        const group: DBGroup = await firestoreCtrl.getGroup(group_id);
+        if (!isInGroupArea(location, group)) {
+          alert("You need to be in the group's area to create a challenge");
```

Comment:
Great feature to implement ! I love this

Reply:
Thank you! It took a long time